File: sdk/src/main/java/com/meniga/sdk/webservices/requests/GetChallengeHistory.java
package com.meniga.sdk.webservices.requests;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

/**
 * Copyright 2017 Meniga Iceland Inc.
 */
public class GetChallengeHistory extends QueryRequestObject {
	public UUID id;
	public int skip;
	public int take;

	@Override
	public long getValueHash() {
		int result = id != null ? id.hashCode() : 0;
		result = 31 * result + skip;
		result = 31 * result + take;
		return result;
	}

	@Override
	public Map<String, String> toQueryMap() {
		Map<String, String> map = new HashMap<>();
		map.put("skip", Integer.toString(skip));
		map.put("take", Integer.toString(take));
		return map;
	}
}

File: sdk/src/main/java/com/meniga/sdk/webservices/challenge/GetChallengeHistory.kt
/*
 * Copyright 2017-2018 Meniga Iceland Inc.
 */
package com.meniga.sdk.webservices.challenge

import com.meniga.sdk.webservices.requests.QueryRequestObject
import java.util.UUID

internal data class GetChallengeHistory(
        @JvmField var id: UUID,
        @JvmField var skip: Int = 0,
        @JvmField var take: Int? = null
) : QueryRequestObject() {

    override fun getValueHash(): Long = hashCode().toLong()

    override fun toQueryMap(): Map<String, String> =
            mutableMapOf<String, String>().also { queryMap ->
                skip.let { queryMap["skip"] = Integer.toString(it) }
                take?.let { queryMap["take"] = Integer.toString(it) }
            }
}
